optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -574,6 +574,120 @@ the stream.
+Consider the following constraint stream:
+
+[source,java,options="nowrap"]
+----
+    protected Constraint horizontalConflict(ConstraintFactory factory) {
+        return factory
+                .fromUniquePair(Queen.class, equal(Queen::getRowIndex))
+                .penalize("Horizontal conflict", SimpleScore.ONE);
+    }
+----
+
+This is how a simple unit test for this constraint looks, using the Constraint Verifier API:
+
+[source,java,options="nowrap"]
+----
+    private ConstraintVerifier<NQueensConstraintProvider, NQueens> constraintVerifier
+            = ConstraintVerifier.build(new NQueensConstraintProvider(), NQueens.class, Queen.class);
+
+    @Test
+    public void horizontalConflictWithTwoQueens() {
+        Row row1 = new Row(0);
+        Column column1 = new Column(0);
+        Column column2 = new Column(1);
+        Queen queen1 = new Queen(0, row1, column1);
+        Queen queen2 = new Queen(1, row1, column2);
+        constraintVerifier.verifyThat(NQueensConstraintProvider::horizontalConflict)
+                .given(queen1, queen2)
+                .penalizesBy(1);
+    }
+----
+
+This test ensures that the "Horizontal conflict" constraint gives a penalty of `1` when there are two queens on the
+same row.
+Let's take a look at the test in detail.
+The following line creates a shared `ConstraintVerifier` instance, initializing it with the `NQueensConstraintProvider`
+that is being tested, as well as the planning solution (`NQueens`) and the planning entity (`Queen`).
+
+[source,java,options="nowrap"]
+----
+    private ConstraintVerifier<NQueensConstraintProvider, NQueens> constraintVerifier
+            = ConstraintVerifier.build(new NQueensConstraintProvider(), NQueens.class, Queen.class);
+----
+
+The `@Test` annotation indicates that the method is a unit test in a testing framework of your choice.
+Constraint Verifier works with every unit testing framework.
+
+First part of the test prepares the test data.
+In this case, two instances of the `Queen` planning entity and their dependencies (`Row`, `Column`):
+
+[source,java,options="nowrap"]
+----
+        Row row1 = new Row(0);
+        Column column1 = new Column(0);
+        Column column2 = new Column(1);
+        Queen queen1 = new Queen(0, row1, column1);
+        Queen queen2 = new Queen(1, row1, column2);
+----
+
+Further down, the following code test the constraint itself:
+
+[source,java,options="nowrap"]
+----
+    constraintVerifier.verifyThat(NQueensConstraintProvider::horizontalConflict)
+            .given(queen1, queen2)
+            .penalizesBy(1);
+----
+
+The `verifyThat(...)` call is used to specify a method on the `NQueensConstraintProvider` class which is under test.
+This method must be visible to the test class, which the Java compiler will enforce.
+
+The `given(...)` call is used to enumerate all the facts that the constraint stream will operate on.
+In this case, the `queen1` and `queen2` instances previously created.
```

Comment:
In this case, the `given(...)` operates on the  `queen1` and `queen2` instances previously created.

Not sure if operates is the correct verb. Substitute or rewrite if need. Just make sure you do include a verb.